apps/gk_field_multib.c

```diff
@@ -0,0 +1,197 @@
+#include <gkyl_alloc.h>
+#include <gkyl_array_ops.h>
+#include <gkyl_bc_basic.h>
+#include <gkyl_dg_eqn.h>
+#include <gkyl_util.h>
+#include <gkyl_gyrokinetic_multib_priv.h>
+
+#include <assert.h>
+#include <float.h>
+#include <time.h>
+
+static void
+insert_below(int* arr, int *n, int new_val)
+{
+  int temp_arr[GKYL_MAX_BLOCKS] = {-1};
+  for (int i = 0; i<*n; i++) {
+    temp_arr[i+1] = arr[i];
+  }
+  temp_arr[0] = new_val;
+  *n = *n+1;
+  for (int i = 0; i<*n; i++) {
+    arr[i] = temp_arr[i];
+  }
+
+}
+
+static void
+insert_above(int* arr, int *n, int new_val)
+{
+  arr[*n] = new_val;
+  *n = *n+1;
+}
+
+// This function should be called in a loop over num blocks local
+void
+set_crossz_idxs(struct gkyl_gyrokinetic_multib_app *mba, struct gkyl_gyrokinetic_app *app, int myidx, int* crossz_blocks, int* num_blocks){
```

Comment:
this doesn't seem like a trivial function. Please include a good description in a comment at the beginning of it.